src/components/InstrumentPicker/__tests__/InstrumentListItem.spec.tsx

```diff
@@ -0,0 +1,322 @@
+import {shallow} from 'enzyme';
+import * as React from 'react';
+import {AssetModel, InstrumentModel} from '../../../models';
+import {formattedNumberWithDashes} from '../../../utils/localFormatted/localFormatted';
+import {colors} from '../../styled';
+import InstrumentListItem from '../InstrumentListItem';
+
+describe('<InstrumentListItem>', () => {
+  let baseAsset: AssetModel;
+  let instrument: InstrumentModel;
+  let inactive: boolean;
+  let isAuth: boolean;
+  let onPick: (instrument: InstrumentModel) => void;
+
+  const getTestIntrumentListItem = () => {
+    return (
+      <InstrumentListItem
+        baseAsset={baseAsset}
+        instrument={instrument}
+        inactive={inactive}
+        isAuth={isAuth}
+        onPick={onPick}
+      />
+    );
+  };
+
+  beforeEach(() => {
+    baseAsset = new AssetModel({
+      id: 'BTC',
+      name: 'BTC',
+      canBeBase: true
+    });
+
+    instrument = new InstrumentModel({
+      id: 'BTCUSD',
+      baseAsset,
+      quoteAsset: new AssetModel({id: 'USD'}),
+      ask: 100
+    });
+
+    inactive = true;
+    isAuth = true;
+    onPick = jest.fn();
+  });
+
+  it('should render table row', () => {
+    const wrapper = shallow(getTestIntrumentListItem());
+    expect(wrapper.find('tr')).toHaveLength(1);
+  });
+
+  it('should contain 5 table divisions', () => {
+    const wrapper = shallow(getTestIntrumentListItem());
+    expect(wrapper.find('td')).toHaveLength(5);
+  });
+
+  describe('table divisions', () => {
+    it('first table division should contain instrument display name', () => {
+      const wrapper = shallow(getTestIntrumentListItem());
+      const td = wrapper.find('td').at(0);
+      expect((td.props() as any).title).toBe(instrument.displayName);
+      expect(td.html()).toContain(instrument.displayName);
+    });
+
+    describe('second table division', () => {
+      it('should contain child with color prop with white color', () => {
+        instrument.price = 123;
+        const wrapper = shallow(getTestIntrumentListItem());
+        const child = wrapper
+          .find('td')
+          .at(1)
+          .childAt(0);
+        expect((child.props() as any).color).toBe(colors.white);
+      });
+
+      it('should contain child with color prop with grey color', () => {
+        instrument.price = 0;
+        const wrapper = shallow(getTestIntrumentListItem());
+        const child = wrapper
+          .find('td')
+          .at(1)
+          .childAt(0);
+        expect((child.props() as any).color).toBe(colors.lightGrey);
+      });
+
+      it('should contain child with num prop with formatted number', () => {
+        const wrapper = shallow(getTestIntrumentListItem());
+        const child = wrapper
+          .find('td')
+          .at(1)
+          .childAt(0);
+        expect((child.props() as any).num).toBe(
+          formattedNumberWithDashes(
```

Comment:
Not a very good test. Tests should check that result of the function is the same as result of calculation, that test will check that dev did not changed the code.